master/internal/webhooks/postgres_webhook.go

```diff
@@ -272,6 +280,121 @@ func (l *WebhookManager) deleteWebhook(ctx context.Context, id WebhookID) error
 	return nil
 }
 
+func handleCustomTriggerData(ctx context.Context, data CustomTriggerData, experimentID int, trialID *int) error {
+	var m struct {
+		bun.BaseModel `bun:"table:experiments"`
+		model.Experiment
+		ConfigBytes []byte `bun:"config"`
+	}
+	err := db.Bun().NewSelect().Model(&m).ExcludeColumn("username").Where("id = ?", experimentID).Scan(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting experiment from id %d: %w", experimentID, err)
+	}
+
+	activeConfig, err := expconf.ParseAnyExperimentConfigYAML(m.ConfigBytes)
+	if err != nil {
+		return fmt.Errorf("error parsing experiment config: %w", err)
+	}
+	integrationsConfig := activeConfig.Integrations()
+	if integrationsConfig == nil {
+		return nil
+	}
+	webhookConfig := integrationsConfig.Webhooks
+	if webhookConfig == nil || webhookConfig.Exclude {
+		return nil
+	}
+
+	workspaceID, err := experiment.GetWorkspaceFromExperiment(ctx, &m.Experiment)
+	if err != nil {
+		return fmt.Errorf("error getting workspace from experiment : %w", err)
+	}
+
+	var es []Event
+	if webhookConfig.WebhookID != nil {
+		for _, webhookID := range *webhookConfig.WebhookID {
+			webhook, err := GetWebhook(ctx, webhookID)
+			if err != nil {
+				return fmt.Errorf("error getting webhook from id %d: %w", webhookID, err)
+			}
+			err = generateEventForCustomTrigger(
+				ctx, &es, webhook.Triggers, webhook.WebhookType, webhook.URL, m.Experiment, activeConfig, data, trialID)
+			if err != nil {
+				return fmt.Errorf("error genrating event %d %+v: %w", webhookID, webhook, err)
+			}
+		}
+	}
+	if webhookConfig.WebhookName != nil {
+		for _, webhookName := range *webhookConfig.WebhookName {
+			webhook, err := getWebhookByName(ctx, webhookName, int(workspaceID))
+			if err != nil {
+				return fmt.Errorf("error getting webhook from name %s: %w", webhookName, err)
+			}
+			if webhook == nil {
+				continue
+			}
+			err = generateEventForCustomTrigger(
+				ctx, &es, webhook.Triggers, webhook.WebhookType, webhook.URL, m.Experiment, activeConfig, data, trialID)
+			if err != nil {
+				return fmt.Errorf("error genrating event %s %+v: %w", webhookName, webhook, err)
+			}
+		}
+	}
+
+	if len(es) == 0 {
+		return nil
+	}
+
+	if _, err := db.Bun().NewInsert().Model(&es).Exec(ctx); err != nil {
+		return fmt.Errorf("report experiment state changed inserting event trigger: %w", err)
```

Comment:
would the error message be something like `handle custom trigger data inserting events`?